### PR TITLE
Reimplementation import export json

### PR DESCRIPTION
**Pull request by kobe1104**
This PR is extracted from https://github.com/nasa/openmct/pull/3816
The test is split to a separate PR: https://github.com/nasa/openmct/pull/4225

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?

### Author Checklist

* [x] Changes address original issue?
* [ ] Unit tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue?

### Testing instructions
1. Export any domain objects by right clicking it and select 'export as JSON'
2. Right click a folder and import the objects back by selecting 'import from JSON'
3. The object should be imported and exported without any errors
4. The object that's been imported should be exactly the same as the one that's exported

### CODE REVIEW

**Comment by kobe1104** [comment]
@nikhilmandlik I've removed all the forms changes and kept only the import export plugins. I also removed the old 'import-export' folder.

File: src/plugins/exportAsJSONAction/ExportAsJSONAction.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+        this.externalIdentifiers = [];
+        this.tree = {};
+        this.calls = 0;
+        this.idMap = {};
+
+        this.exportService = new ExportService(saveAs);
+    }
+
+    // Public
```
Comment (nikhilmandlik) [inline src/plugins/exportAsJSONAction/ExportAsJSONAction.js:46]: sorry I missed this one, can we add JsDoc

File: src/plugins/exportAsJSONAction/ExportAsJSONAction.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    invoke(objectpath) {
+        const root = objectpath[0];
+        this.root = JSON.parse(JSON.stringify(root));
+        const rootId = this._getId(this.root);
+        this.tree[rootId] = this.root;
+
+        this._write(this.root);
+    }
+
+    // Private
```
Comment (nikhilmandlik) [inline src/plugins/exportAsJSONAction/ExportAsJSONAction.js:62]: add JsDoc

File: src/plugins/importFromJSONAction/ImportFromJSONAction.js [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+        this.key = 'import.JSON';
+        this.description = '';
+        this.cssClass = "icon-import";
+        this.group = "json";
+        this.priority = 2;
+
+        this.openmct = openmct;
+    }
+
+    // Public
```
Comment (nikhilmandlik) [inline src/plugins/importFromJSONAction/ImportFromJSONAction.js:38]: same add JsDoc

File: src/plugins/importFromJSONAction/ImportFromJSONAction.js [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+        this._showForm(objectPath[0], objectPath[1]);
+    }
+
+    onSave(object, changes, parent) {
+        const selectFile = changes.selectFile;
+        const objectTree = selectFile.body;
+        this._importObjectTree(object, JSON.parse(objectTree));
+    }
+
+    // Private
```
Comment (nikhilmandlik) [inline src/plugins/importFromJSONAction/ImportFromJSONAction.js:60]: add JsDoc

**Comment by kobe1104** [comment]
Hi @nikhilmandlik I've added the JSdoc to them. I added mostly the data types. Let me know if there's anything else.

File: src/plugins/exportAsJSONAction/ExportAsJSONAction.js [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+import ExportService from '../../../platform/exporters/ExportService.js';
```
Comment (akhenry) [inline src/plugins/exportAsJSONAction/ExportAsJSONAction.js:22]: Awesome that we can just reuse the ExportService as is, but this needs to be moved under `src/`. The whole `platform/` directory is going to go away.

File: src/plugins/importFromJSONAction/ImportFromJSONAction.js [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+                            text: 'Select File...',
+                            validate: this._validateJSON,
+                            type: 'application/json'
+                        }
+                    ]
+                }
+            ]
+        };
+
+        this.openmct.forms.showForm(formStructure, {
```
Comment (akhenry) [inline src/plugins/importFromJSONAction/ImportFromJSONAction.js:220]: @nikhilmandlik Remind me why all forms need a `domainObject` and a `parentDomainObject`? In fact, in this case it doesn't seem as though any domain objects should be needed?

**Review by akhenry — COMMENTED** [review]
This looks really great @kobe1104. It just needs some tests.

**Review by akhenry — CHANGES_REQUESTED** [review]
sdfsdfdsfsdfsdfsdf

**Comment by nikhilmandlik** [comment]
@akhenry just a test comment 😄

**Comment by kobe1104** [comment]
Hi @akhenry I've created a separated PR for the test: https://github.com/nasa/openmct/pull/4225

**Comment by kobe1104** [comment]
Hi @akhenry The new import/export JSON are moved under /src and I've removed the old import/export actions from platform. Also the comments of '// Private' have been removed.

**Comment by akhenry** [comment]
@kobe1104 @nikhilmandlik to resolve merge conflicts.

**Comment by kobe1104** [comment]
@akhenry @nikhilmandlik I've fixed the conflict

File: src/exporters/JSONExporter.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+import {saveAs} from 'file-saver/FileSaver';
+
+class JSONExporter {
+    export(obj, options) {
+        let filename = (options && options.filename) || "test-export.json";
```
Comment (nikhilmandlik) [inline src/exporters/JSONExporter.js:27]: test-export.json?

**Review by nikhilmandlik — APPROVED** [review]
Small change else looks good

**Pull request merged**